examples/xgboost/custom/xgboost_bagging_aggregator.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+
+from nvflare.apis.dxo import DXO, DataKind, from_shareable
+from nvflare.apis.fl_constant import FLContextKey, ReservedKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.app_constant import AppConstants
+
+
+class XGBoostBaggingAggregator(Aggregator):
+    def __init__(
+        self,
+        expected_data_kind: DataKind = DataKind.WEIGHTS,
+    ):
+        """Perform bagging aggregation for XGBoost trees.
+
+        The trees are pre-weighted during training.
+        Bagging aggregation simply add the new trees to existing global model.
+
+        Args:
+            expected_data_kind (Union[DataKind, Dict[str, DataKind]]):
+                DataKind for DXO. Defaults to DataKind.WEIGHTS
+                Indicating the tree representation given by XGBoost
```

Comment:
since we only support `DataKind.WEIGHTS` now.
We can remove this argument.

Reply:
changed, and as Yan suggested, add a new data_kind called "XGB_MODEL"